lib/neo4j/core/cypher_session/adaptors.rb

```diff
@@ -52,68 +57,71 @@ def query
             end
           end
 
-          def query(*args)
-            queries { append(*args) }[0]
+          def query(session, *args)
+            options = (args.size == 3 || (args[0].is_a?(::Neo4j::Core::Query) && args.size == 2)) ? args.pop : {}
```

Comment:
Why does size matter?

:trollface: 

But really, is there another way to express this that can help the reader understand what this means?